app/src/main/java/com/github/polybooks/core/database/implementation/OLBookDatabase.kt

```diff
@@ -182,9 +183,11 @@ class OLBookDatabase(private val url2json : (String) -> CompletableFuture<JsonEl
     @RequiresApi(Build.VERSION_CODES.N)
     private fun parseBook(jsonBook: JsonElement): Book {
         val jsonBookObject = asJsonObject(jsonBook);
-        val title = getJsonField(jsonBookObject, TITLE_FIELD_NAME)
+        val title = getJsonField(jsonBookObject, TITLE_FIELD_NAME1)
+                .map { Optional.of(it) }
+                .orElseGet { getJsonField(jsonBookObject, TITLE_FIELD_NAME2) }
                 .map { parseTitle(it) }
-                .orElseThrow(cantParseException(TITLE_FIELD_NAME))!!
+                .orElseThrow(cantParseException("$TITLE_FIELD_NAME1 or $TITLE_FIELD_NAME2"))!!
```

Comment:
I don't really understand what is written, but seems ok to me?

Reply:
Mapping the first Option to an Option of itself is just a hack (because java's Optional class is limited) to compute the OR of two options: If the first option is None, try with the second one, and if the second one is also None, return None.